File: app/src/main/java/fr/nspu/dev/recordvoicelearning/utils/QuestionToAnswerEnum.java
package fr.nspu.dev.recordvoicelearning.utils;

/**
 * Created by nspu on 18-02-23.
 */

public enum QuestionToAnswerEnum {

    QUESTION_TO_ANSWER(true),
    ANSWER_TO_QUESTION(false);

    Boolean questionToAnswer = true;


    QuestionToAnswerEnum(boolean b) {
        questionToAnswer = b;
    }

    public boolean toBoolean() {
        return questionToAnswer;
    }

    public static QuestionToAnswerEnum toQuestionToAnswerEnum(Boolean b) {
        if (b == null || b) {
            return QUESTION_TO_ANSWER;
        } else {
            return ANSWER_TO_QUESTION;
        }
    }
}

File: app/src/main/java/fr/nspu/dev/recordvoicelearning/utils/QuestionToAnswerEnum.kt
package fr.nspu.dev.recordvoicelearning.utils

/**
 * Created by nspu on 18-02-23.
 */

enum class QuestionToAnswerEnum private constructor(b: Boolean) {

    QUESTION_TO_ANSWER(true),
    ANSWER_TO_QUESTION(false);

    internal var questionToAnswer: Boolean? = true


    init {
        questionToAnswer = b
    }

    fun toBoolean(): Boolean = questionToAnswer!!

    companion object {
        fun toQuestionToAnswerEnum(b: Boolean?): QuestionToAnswerEnum? = if (b == null || b) QUESTION_TO_ANSWER else ANSWER_TO_QUESTION
    }
}
